101-application-gateway-rewrite/azuredeploy.json

```diff
@@ -0,0 +1,312 @@
+      }
+    },
+    "MSIPort": {
+      "type": "int",
+      "defaultValue": 50342,
+      "metadata": {
+        "description": "Default port for msi extension"
+      }
+    },
+    "KeyVaultVersion": {
+      "type": "string",
+      "defaultValue": "2018-02-14-preview",
+      "metadata": {
+        "description": "API version for key vault resources and access policy resources"
+      }
+    },
+    "Secrets": {
+      "type": "array",
+      "metadata": {
+        "description": "An array of json objects like this : {'name':name, 'value':value}"
+      }
+    }
+  },
+  "variables": {
+    "applicationGatewayName": "[concat(resourceGroup().name, '-appgw')]",
+    "idName": "identity1",
+    "keyVaultName": "resource-kv",
+    "publicIPAddressName": "[concat(resourceGroup().name, '-publicIp')]",
+    "virtualNetworkName": "[concat(resourceGroup().name, '-vnet')]",
+    "subnetName": "subnet1",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
+    "applicationGatewayID": "[resourceId('Microsoft.Network/applicationGateways',variables('applicationGatewayName'))]",
+    "apiVersion": "2018-10-01",
+    "identityID": "[resourceId('Microsoft.ManagedIdentity/userAssignedIdentities',variables('idName'))]",
+    "keyVaultID": "[resourceId('Microsoft.KeyVault/vaults',variables('keyVaultName'))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.ManagedIdentity/userAssignedIdentities",
+      "name": "[variables('idName')]",
+      "apiVersion": "2015-08-31-PREVIEW",
+      "location": "[resourceGroup().location]"
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults",
+      "name": "[variables('keyVaultName')]",
+      "apiVersion": "[parameters('KeyVaultVersion')]",
+      "location": "[resourceGroup().location]",
+      "dependsOn": [
+        "[concat('Microsoft.ManagedIdentity/userAssignedIdentities/', variables('idName'))]"
+      ],
+      "properties": {
+        "sku": {
+          "name": "standard",
+          "family": "A"
+        },
+        "tenantId": "[reference(variables('identityID'), parameters('MSIVersion')).tenantId]",
+        "accessPolicies": [
+          {
+            "tenantId": "[reference(variables('identityID'), parameters('MSIVersion')).tenantId]",
+            "objectId": "[reference(variables('identityID'), parameters('MSIVersion')).principalId]",
+            "permissions": {
+              "secrets": [
+                "get"
+              ]
+            }
+          }
+        ],
+        "enableSoftDelete": true
+      }
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults/secrets",
+      "dependsOn": [
+        "[resourceId('Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
+      ],
+      "apiVersion": "[parameters('KeyVaultVersion')]",
```

Comment:
all the apiVersion properties on all resources need to be literal values

Reply:
Done.